src/test/components/login/login.spec.js

```diff
@@ -95,179 +108,63 @@ describe('Login controller', () => {
-      controller.doTheLogin();
-      expect(controller.passphrase).to.equal('glow two glimpse camp aware tip brief confirm similar code float defense');
+  describe('passConfirmSubmit()', () => {
+    it('sets this.phassphrase as this.input_passphrase processed by normalizer', () => {
+      controller.input_passphrase = '\tTEST  PassPHrASe  ';
+      controller.passConfirmSubmit();
+      expect(controller.passphrase).to.equal('test passphrase');
     });
 
-    it('calls this.reset()', () => {
-      controller.input_passphrase = testPassphrase;
-      const spy = sinon.spy(controller, 'reset');
-      controller.doTheLogin();
+    it('calls Passphrase.normalize()', () => {
+      const spy = sinon.spy(Passphrase, 'normalize');
+      controller.passConfirmSubmit();
       expect(spy).to.have.been.calledWith();
     });
 
     it('sets timeout with this.onLogin', () => {
       controller.input_passphrase = testPassphrase;
       const spy = sinon.spy(controller, '$timeout');
-      controller.doTheLogin();
+      controller.passConfirmSubmit();
       expect(spy).to.have.been.calledWith(controller.onLogin);
     });
   });
 
-  describe('$scope.constructor()', () => {
-    it.skip('sets $watch on $ctrl.input_passphrase to keep validating it', () => {
-      // Skipped because it doesn't work
-      const spy = sinon.spy(controller.$scope, '$watch');
-      controller.constructor();
-      expect(spy).to.have.been.calledWith('$ctrl.input_passphrase', controller.isValidPassphrase);
-    });
-
-    it.skip('sets $watch that sets customFullscreen on small screens', () => {
-    });
-  });
-
-  describe('$scope.simulateMousemove()', () => {
-    it('calls this.$document.mousemove()', () => {
-      const spy = sinon.spy(controller.$document, 'mousemove');
-      controller.simulateMousemove();
-      expect(spy).to.have.been.calledWith();
-    });
-  });
-
-  describe('$scope.setNewPassphrase()', () => {
-    it('opens a material design dialog', () => {
-      const seed = ['23', '34', '34', '34', '34', '34', '34', '34'];
-      const dialogSpy = sinon.spy(controller.$mdDialog, 'show');
-      controller.setNewPassphrase(seed);
-      expect(dialogSpy).to.have.been.calledWith();
-    });
-  });
-
-  describe('$scope.devTestAccount()', () => {
-    it('sets input_passphrase from cookie called passphrase if present', () => {
-      const mock = sinon.mock(controller.$cookies);
-      mock.expects('get').returns(testPassphrase);
-      controller.devTestAccount();
-      expect(controller.input_passphrase).to.equal(testPassphrase);
-    });
-
-    it('does nothing if cooke called passphrase not present', () => {
-      controller.input_passphrase = testPassphrase;
-      const mock = sinon.mock(controller.$cookies);
-
-      mock.expects('get').returns(undefined);
+  describe('devTestAccount()', () => {
+    it('calls passConfirmSubmit with timeoout if a passphrase is set in the cookies', () => {
```

Comment:
Spelling: `calls passConfirmSubmit with timeout if a passphrase is set in the cookies`